.github/workflows/ci.yml

```
@@ -1,126 +1,74 @@
 name: ci
-
 on:
   push:
     branches:
       - master
-      - '2.0'
     paths-ignore:
       - '*.md'
   pull_request:
     paths-ignore:
       - '*.md'
 
+permissions:
+  contents: read
+
 jobs:
   test:
+    name: Test - Node.js ${{ matrix.node-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        name:
-        - Node.js 18.x
-        - Node.js 19.x
-        - Node.js 20.x
-        - Node.js 21.x
-        - Node.js 22.x
-
-        include:
-        - name: Node.js 18.x
-          node-version: "18"
-
-        - name: Node.js 19.x
-          node-version: "19"
-
-        - name: Node.js 20.x
-          node-version: "20"
-
-        - name: Node.js 21.x
-          node-version: "21"
-
-        - name: Node.js 22.x
-          node-version: "22"
+        node-version: [18, 19, 20, 21, 22, 23]
 
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
```

```suggestion
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

It would be better to use the hash of the last commit of the version we want to use in each GitHub action, it's just to avoid opening another PR later.

Example: https://github.com/expressjs/compression/pull/205

---

I don't recommend pinning GitHub actions with commit hashes, as it introduces a maintenance burden. Without an automatic dependency update system in place, manually updating these pinned hashes becomes cumbersome. In my opinion, using version tags like v4 ensures stability while allowing for easier maintenance and updates.

---

As soon as a dependency update tool is implemented, I fully support pinning to commit hashes to ensure more controlled and reliable updates.

---

I would like to know your opinions @UlisesGascon @wesleytodd

---

I would not bother pinning any actions from GitHub itself. The security risk there is basically non-existent. If those were ever compromised then that would be a world news level exploit. I am pretty sure the only reason this is recommend is that 3rd party actions are much more susceptible, and GitHub wants to cover their bases, hence the recommendation in the docs.

---

We need to do an RFC to align on this I think. We have typically considered the core platform ones "safe" in this reagard and the third party ones "unsafe". That said, I think we should have a more serious conversation on the pros/cons for the future. For example, I am strongly opposed to locking unless we have an auto update mechanism in place, ideally with some simple auto-merge logic (passed builds and one approval?).

For today, I think it is good to keep the ranges because that is also what we are doing in most repos.

---

Lets call this resolved and move the conversation into an RFC so we can merge this improvement.